v2/inc/c_logging/log_context_property_type_wchar_t_ptr.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_H
+#define LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_H
+
+#include "c_logging/log_context_property_type_if.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...);
+    int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...);
+
+    extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr);
```

Comment:
>extern [](http://example.com/codeflow?start=4&length=6)

why is "extern" needed here? in C all declarations are `extern` default.

Reply:
Ok, now I understand and I like it with `extern`.

I like the explicitness.

Reply:
I don't understand why I paste it. Why do we need extern here?

Reply:
We don't "need"; it would without "extern" equally nice.

But I like it "with" extern - as it makes the intention very clear. We usually don't have extern variables (note:  LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr) expands to a variable name).

The reason why we don't have extern variables (and like 99.99% we don'thave extern variables) is because once they are "extern" they also become global and therefore accesible from ANY module, which would violate encapsulation.

However!! This is a "const" so all writes would fail.

Reply:
make sense thank you for explanation :)